### ISSUE TITLE
Still incompatible with Julia <1.7

### ISSUE THREAD

**Issue by MilesCranmer** [JuliaLang/BugReporting.jl#156]
cc @maleadt unfortunately this package is still broken on Julia <1.7.  Could you instead set the bounds of the Julia versions according to what Julia you have actually tested against? Otherwise when I try to submit bug report for an earlier Julia version, it just downloads the current BugReporting.jl (which appears incompatible with older Julia versions), despite the `[compat]` saying `julia = "1.3"`:

https://github.com/JuliaLang/BugReporting.jl/blob/88ef063c0df5e1ae3018147ffc962ec0163b485a/Project.toml#L29

If the `[compat]` is given correctly then it will download the older version of BugReporting which would let me successfully submit a bug report.

The next bug is that `errormonitor` does not yet exist in Julia 1.6:

```julia
ERROR: UndefVarError: errormonitor not defined
Stacktrace:
  [1] get_upload_params()
    @ BugReporting ~/.julia/packages/BugReporting/NFHSR/src/BugReporting.jl:632
  [2] (::BugReporting.var"#20#22"{Vector{String}})(trace_dir::String)
    @ BugReporting ~/.julia/packages/BugReporting/NFHSR/src/BugReporting.jl:551
  [3] mktempdir(fn::BugReporting.var"#20#22"{Vector{String}}, parent::String; prefix::String)
    @ Base.Filesystem ./file.jl:729
  [4] mktempdir (repeats 2 times)
    @ ./file.jl:727 [inlined]
  [5] make_interactive_report(report_arg::String, ARGS::Vector{String})
    @ BugReporting ~/.julia/packages/BugReporting/NFHSR/src/BugReporting.jl:547
  [6] #invokelatest#2
    @ ./essentials.jl:708 [inlined]
  [7] invokelatest
    @ ./essentials.jl:706 [inlined]
  [8] report_bug(kind::String)
    @ InteractiveUtils /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/InteractiveUtils/src/InteractiveUtils.jl:391
  [9] exec_options(opts::Base.JLOptions)
    @ Base ./client.jl:233
 [10] _start()
    @ Base ./client.jl:485
```

**Comment by MilesCranmer** [comment]
I was able to get it working with BugReporting.jl version 0.2.7

It looks like every version since then, i.e., v0.2.8-0.3.3, are all incompatible with Julia <1.7 (or 1.7 as well; but I've only tested 1.6 thus far).

@maleadt Because of this incompatibility, could those compatibilities be yanked from the registry for Julia <1.7? Otherwise you would need to know specifically what BugReporting.jl version to install, and couldn't rely on `--bug-report=rr` to get a working one.
